algorithms/integration/integrator.py

```diff
@@ -1164,16 +1183,94 @@ def integrate(self):
 
         # Create the data processor
         executor = IntegratorExecutor(self.experiments, profile_fitter)
-        processor = build_processor(
-            self.ProcessorClass,
-            self.experiments,
-            self.reflections,
-            self.params.integration,
-        )
-        processor.executor = executor
 
-        # Process the reflections
-        self.reflections, _, time_info = processor.process()
+        # determine the max memory needed during integration
+        def _determine_max_memory_needed(experiments, reflections):
+            max_needed = 0
+            for imageset in experiments.imagesets():
+                try:
+                    frame0, frame1 = imageset.get_array_range()
+                except Exception:
```

Comment:
can this be narrowed down to expected exceptions?

Reply:
Looks like only a DXTBX_ASSERT can be raised by the imageset if there is no scan. I believe this is exposed as a RuntimeError in python, so i'll update this and add a comment.